src/Compilers/Core/CodeAnalysisTest/Diagnostics/OperationTests.cs

```diff
@@ -272,5 +272,75 @@ private void TestGetFlowGraphInvalidArgumentCore(string argumentExceptionMessage
                 Assert.Equal(new ArgumentException(argumentExceptionMessage, "methodBody").Message, ex.Message);
             }
         }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void TestControlFlowGraphCreateFromSyntax()
+        {
+            var source = @"
+class C
+{
+    void M(int x)
+    {
+        x = 0;
+    }
+}";
+            var tree = CSharpSyntaxTree.ParseText(source);
+            var options = tree.Options.WithFeatures(new[] { new KeyValuePair<string, string>("flow-analysis", "true") });
+            tree = tree.WithRootAndOptions(tree.GetRoot(), options);
+            var compilation = CSharpCompilation.Create("c", new[] { tree });
+            var model = compilation.GetSemanticModel(tree, ignoreAccessibility: true);
+            var methodBodySyntax = tree.GetCompilationUnitRoot().DescendantNodes().OfType<BaseMethodDeclarationSyntax>().Last();
+
+            // Verify ArgumentNullException
+            try
+            {
+                _ = ControlFlowGraph.Create(node: null, model);
+            }
+            catch (ArgumentNullException ex)
+            {
+                Assert.Equal(ex.Message, new ArgumentNullException("node").Message);
+            }
+
+            try
+            {
+                _ = ControlFlowGraph.Create(methodBodySyntax, semanticModel: null);
```

Comment:
>ControlFlowGraph.Create(methodBodySyntax, semanticModel: null); [](start = 20, length = 63)

It would be good to test situation when syntax node and semantic model do not come from the same tree.

Reply:
Actually, we cannot test this code path as GetOperation catches all exceptions and fires a debug assert: [http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/SemanticModel.cs,81](http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/SemanticModel.cs,81). We can probably clean this up/add tests in a separate PR.

---
In reply to: [197973051](https://github.com/dotnet/roslyn/pull/28079#discussion_r197973051) [](ancestors = 197973051)